### DISCUSSION TITLE
Trouble using filter() with relationship

### DISCUSSION THREAD

**Discussion by ConfSlay** [sqlalchemy/sqlalchemy#11259] [Usage Questions]
Hello there, 

Lets say i have this really simple usecase:
```py
class Task:
    requestor = db.relationship(
        "User",
        back_populates="requested_tasks",
        foreign_keys=[requestor_id],
    )
  ...

class User:
sub_type = db.relationship("UserSubType", back_populates="users")
...

class UserSubType:
...
```
Im trying to do :
```
         prj_models.Task.query
        .filter( **_Task.requestor.sub_type.id.in_(some_ids)_**  )
        .all()
```
 But im getting this error : 
```
AttributeError: Neither 'InstrumentedAttribute' object nor 'Comparator' object associated with Task.requestor has an attribute 'sub_type'
```
I tried some trials of joinedload but nothing really works and i would like to know how you guys would apprehend this problem, maybe im missing an important concept of sqlalchemy. Obviously, the attribute exists, i have no problem printing his value with python if i make a request without filter. 

Thanks a lot ! :)

**Comment by CaselIT** [comment]
Hi,

You cannot access an attribute of a relationship target class like that. You would use something like https://docs.sqlalchemy.org/en/20/orm/internals.html#sqlalchemy.orm.RelationshipProperty.Comparator.any or https://docs.sqlalchemy.org/en/20/orm/internals.html#sqlalchemy.orm.RelationshipProperty.Comparator.has

**Reply by ConfSlay** [reply]
Indeed !! Thanks a lot, have a good day :)
